lib/charms/vault_k8s/v0/juju_facade.py

```diff
@@ -543,6 +638,46 @@ def get_storage_location(self, storage_name: str) -> Path:
             raise NoSuchStorageError(f"Storage {storage_name} not found")
         return storages[storage_name][0].location
 
+    def get_binding(
+        self,
+        relation_name: str,
+        relation: Relation | None = None,
+        relation_id: int | None = None,
+    ) -> Binding | None:
+        """Get the binding for the given relation."""
+        if not relation and not relation_id:
+            raise ValueError("Either relation or relation_id must be provided")
+        relation = (
+            (
+                self.get_relation_by_id(relation_name, relation_id)
+                if relation_id
+                else self.get_relation_by_name(relation_name)
+            )
+            if not relation
+            else relation
+        )
+        return self.charm.model.get_binding(relation)
+
+    def get_egress_subnets(
+        self,
+        relation_name: str,
+        relation_id: int | None = None,
+        relation: Relation | None = None,
+    ) -> List[str]:
+        """Get the list of egress subnets for the given relation.
+
+        This returns how units on the relation will see the charm connecting from.
```

Comment:
An example of what this subnet look like would be useful, especially since it's a string.